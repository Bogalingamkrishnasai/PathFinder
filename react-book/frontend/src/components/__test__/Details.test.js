import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Details from '../Details';

// Mock the Outlet component
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Outlet: jest.fn(() => <div data-testid="outlet-content">Outlet Content</div>),
}));

describe('Details Component Snapshots', () => {
  const mockItems = [
    { itemId: '1', title: 'Item 1', imageId: 'img1' },
    { itemId: '2', title: 'Item 2', imageId: 'img2' }
  ];

  test('matches sidebar snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Details items={mockItems} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('matches outlet snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Details items={mockItems} />}>
            <Route index element={<div>Default Content</div>} />
          </Route>
        </Routes>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});