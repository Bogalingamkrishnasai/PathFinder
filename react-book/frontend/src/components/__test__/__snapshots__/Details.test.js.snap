// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component Snapshots matches sidebar snapshot 1`] = `
<DocumentFragment>
  <div
    class="details-component"
  >
    <div
      data-testid="outlet-content"
    >
      Outlet Content
    </div>
    <div
      class="details-component-sidebar"
    >
      <a
        class="thumbnail-component"
        href="/details/1"
      >
        <div>
          <img
            alt="Item 1"
          />
        </div>
        <p>
          Item 1
        </p>
      </a>
      <a
        class="thumbnail-component"
        href="/details/2"
      >
        <div>
          <img
            alt="Item 2"
          />
        </div>
        <p>
          Item 2
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
